File: 2.Traverse/Python/34.Traverse2D.py
# Traverse 2d array
def Traverse_arr(arr,N,M):
    matrix = []
    for col in range(M-1,-1,-1):
        for row in range(N):
            matrix.append(arr[row][col])

    return " ".join(map(str,matrix))



arr = [ [1,8,9], [2,7,10] , [3,6,11], [4,5,12]] 

N = 4
M = 3
res = Traverse_arr(arr,N,M)
print(res)

# Output:-9 10 11 12 8 7 6 5 1 2 3 4

File: 2.Traverse/Javascript/34.Traverse2D.js
function traverseArray(arr, N, M){
    let res = []
    
    for(let j=M-1;j>=0;j--){
        
        for(let i=N-1;i>=0;i--){
            res.push(arr[i][j])
        }
    }

    return res.join(' ')
}

let arr = [
    [1, 8, 9],
    [2, 7, 10],
    [3, 6, 11],
    [4, 5, 12]
];

let N = 4;
let M = 3;

let res = traverseArray(arr, N, M);
console.log(res);  // Output: 12 11 10 9 5 6 7 8 4 3 2 1